src/org/way2it/data_structures/linked_list/LinkedList.java

```diff
@@ -1,48 +1,207 @@
+            Node<T> next;
+            Node<T> indexVal = new Node<>();
+            indexVal.value = value;
+            int count = 0;
+            while (current != null) {
+                if(count == index - 1) {
+                    next = current.next;
+                    current.next = indexVal;
+                    indexVal.next = next;
+                    length++;
+                    return;
+                }
+                else {
+                    current = current.next;
+                    count++;
+                }
+            }
+        }
+    }
 
     // Should return true if value exists in this list, false - otherwise
-    public boolean contains(String value) {
+    public boolean contains(T value) {
         // TODO implement me
+        if(isEmpty()){
+            return false;
+        }
+        Node<T> current = head;
+        while (current != null) {
+            if (current.value.equals(value)) {
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
+    
+    // This method can return the value of List by index
+    public T get(int index) {
+        if (isEmpty() | index >= length) {
+            return null;
+        }
+        Node<T> current = head;
+        int count = 0;
+        while (current != null) {
+            if(count == index) {
+                return current.value;
+            } else {
+                current = current.next;
+                count++;
+            }
+        }
+        return null;
+    }
+    
+    //This method return index of value in List
+    public int getIndex(T value) {
+        if (isEmpty()) {
+            return -1;
+        }
+        Node<T> current = head;
+        int count = 0;
+        while (current != null) {
+            if(current.value.equals(value)) {
+                return count;
+            }else {
+                current = current.next;
+                count++;
+            }
+        }
+        return -1;
+    }
 
     // Should remove first occurrence of the specified value from this list and decrement the length
     // If value existed and was removed - return true, false - otherwise
-    public boolean remove(String value) {
+    public boolean remove(T value) {
         // TODO implement me
+        if (isEmpty()) {
+            return false;
+        }
+        if (head.value.equals(value)) {
+            head = head.next;
+            length--;
+            return true;
+        }
+        Node<T> current = head;
+        while (current.next != null) {
+            if (current.next.value.equals(value)) {
+                current.next = current.next.next;
+                length--;
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
+    
+    //This method remove value in List by its index
+    public void removeByIndex(int index) {
+        remove(get(index));
```

Comment:
With such implementation you iterate through all list twice: first time to find a value and second time to find it again and delete. That's not really good for performance.